File: objects-exercises/exercise1.py
class Person: 
    def __init__(self, name, email, phone): 
     self.name = name 
     self.email = email 
     self.phone = phone 
     self.friends = []
     self.greeting_count = 0
     print("{}, {}, {}".format(self.name, self.email, self.phone))

     
    def greet(self, other_person): 
        print('Hello {}, I am {}!'.format(other_person, self.name))
        self.greeting_count += 1

    def print_contact_info(self):
         print(self.name + "'s email: " + self.email + ' ' + self.name + "'s phone number: " + self.phone)
    
    def add_friend(self, friends):
        self.friends.append(friends.name)
    
    def num_friends(self):
        print(len(Jordan.friends))
    
    def __str__(self):
        return 'Person: {}, {}, {}'.format(self.name, self.email, self.phone)
    

Sonny = Person('Sonny', 'sonny@hotmail.com', '483-485-4948')
Jordan = Person('Jordan', 'jordan@hotmail.com', '495-586-3456')

print(Sonny.greeting_count)
Sonny.greet('Jordan')
print(Sonny.greeting_count)
Sonny.greet('Jordan')
print(Sonny.greeting_count)
Jordan.greet('Sonny')

Sonny.print_contact_info()

Jordan.add_friend(Sonny)

Jordan.num_friends()

print(str(Jordan))

File: javascript-exercises/week5/tuesday/exercise1.js
function Person(name, email, phone) { 
    this.name = name; 
    this.email = email; 
    this.phone = phone; 
} 

Person.prototype.greet = function(other) { 
    console.log('Hello ' + other.name + ', I am ' + this.name + '!'); 
};

var sonny = new Person('Sonny', 'sonny@hotmail.com', '483-485-4948');
var jordan = new Person('Jordan', 'jordan@aol.com', '495-586-3456');

sonny.greet(jordan);
jordan.greet(sonny);

console.log(sonny.email, sonny.phone);
console.log(jordan.email, jordan.phone);

//Card Constructor
function Card(point, suit){
    this.point = point;
    this.suit = suit;
}


var myCard = new Card(5, 'diamonds');

console.log(myCard.point);
console.log(myCard.suit);

Card.prototype.getImageUrl = function(){
    return(`images/${myCard.point}_of_${myCard.suit}.png`);
}
console.log(myCard.getImageUrl());

//hand constructor
function Hand(){
    this.myCardsInHand = [];
}

Hand.prototype.addCard = function(card){
    this.myCardsInHand.push(card);
}

Hand.prototype.getPoints = function(){
    var self = this;
    var points = self.myCardsInHand.map(function(e){
        return e.point;
    })
    // points.reduce(function(a, b){
    //     return a + b;
    // }, 0)
    // console.log(points);
    let count = 0;
    points.forEach(function(index){
        count = count + index;
    })
    console.log(count);
}
var myHand = new Hand();
myHand.addCard(new Card(5, 'diamond'));
myHand.addCard(new Card(13, 'spades'));
myHand.getPoints();